policies.go

```diff
@@ -42,3 +48,184 @@ type SimpleRetryPolicy struct {
 func (s *SimpleRetryPolicy) Attempt(q RetryableQuery) bool {
 	return q.Attempts() <= s.NumRetries
 }
+
+//HostSelectionPolicy is an interface for selecting
+//the most appropriate host to execute a given query.
+type HostSelectionPolicy interface {
+	//SetHosts notifies this policy of the current hosts in the cluster
+	SetHosts(hosts []HostInfo)
+	//SetPartitioner notifies this policy of the current token partitioner
+	SetPartitioner(partitioner string)
+	//Pick returns an iteration function over selected hosts
+	Pick(*Query) NextHost
+}
+
+//NextHost is an iteration function over picked hosts
+type NextHost func() *HostInfo
+
+//NewRoundRobinHostPolicy is a round-robin load balancing policy
+func NewRoundRobinHostPolicy() HostSelectionPolicy {
+	return &roundRobinHostPolicy{hosts: []HostInfo{}}
+}
+
+type roundRobinHostPolicy struct {
+	hosts []HostInfo
+	pos   uint32
+	mu    sync.RWMutex
+}
+
+func (r *roundRobinHostPolicy) SetHosts(hosts []HostInfo) {
+	r.mu.Lock()
+	r.hosts = hosts
+	r.mu.Unlock()
+}
+
+func (r *roundRobinHostPolicy) SetPartitioner(partitioner string) {
+	// noop
+}
+
+func (r *roundRobinHostPolicy) Pick(qry *Query) NextHost {
+	pos := atomic.AddUint32(&r.pos, 1)
+	var i uint32 = 0
+	return func() *HostInfo {
```

Comment:
To get better query distribution here I think it will be better to have the function increment `pos` each call